### PR TITLE
Simplify FileData::marks_list_load a bit

### PR DESCRIPTION
**Pull request by qarkai**
Use g_auto macros.
Fix indentation.
Use for loop.
Reduce scope of variables and flatten code.

### CODE REVIEW

File: src/filedata/filedata.cc [hunk 1]
```diff
@@ -2820,48 +2820,43 @@ static void marks_get_files(gpointer key, gpointer value, gpointer userdata)
...
 		if (s_buf[0]=='#') continue;
-			file_path = strtok(s_buf, ",");
-			marks_value = strtok(nullptr, ",");
-			if (isfile(file_path))
+
+		const gchar *file_path = strtok(s_buf, ",");
+		if (!isfile(file_path)) continue;
+
+		const gchar *marks_str = strtok(nullptr, ",");
+		const gint marks_value = atoi(marks_str); // @fixme Skip if marks_value is 0?
```
Comment (qarkai) [inline src/filedata/filedata.cc:2845]: @caclark @xsdg What do you think about this?

**Comment by xsdg** [comment]
Agreed with @caclark 

Also, more broadly, I really like this change.  Thanks for continuing the good work 👍

File: src/filedata/filedata.cc (hunk 1 above)
Comment (qarkai) [reply src/filedata/filedata.cc]: Ok, added comment.

**Comment by xsdg** [comment]
LGTM

**Comment by caclark** [comment]
Implemented in commit 9c018368827c1e5650ac057c3d4480169584c801

**Pull request closed without merging**